### ISSUE TITLE
Whether flash attention can be applied in video generation

### ISSUE THREAD

**Issue by Aziily** [jy0205/Pyramid-Flow#222]
Hello, thanks again for your excellent work. But I encounter a problem when generating video.

In 384p model, when `use_flash_attn` is set to False, the model can generate great video like this.

https://github.com/user-attachments/assets/1d565149-69b0-4213-a27f-848ef520b73a

However, when using flash attention, it seems that the model fails to generate one. And will produce a bad result like this 

https://github.com/user-attachments/assets/a67856de-dae3-4f5c-863d-108e6cce0666

Both prompts here are `A dog`. Though I did not maintain the same generator, I think this is not a problem of the seed but something wrong about the flash attention adaptation. I also refer to the training code, and find that in video training you do not use flash attention. Is it true that there are some problems?

**Comment by feifeiobama** [comment]
Flash attention is incompatible with blockwise attention mask at least in the pytorch version we used.

**Comment by Aziily** [comment]
Thank you for your reply.
